core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -341,6 +323,45 @@ public KeyStore getPrimaryKeyStore() throws Exception {
         }
     }
 
+    /**
+     * Load the requested tenant keystore from registry.
+     *
+     * @param keyStoreName  Name of the Tenant KeyStore.
+     * @return Key store object.
+     * @throws Exception    Exception if failed to retrive the given keystore from registry.
+     */
+    private KeyStore getTenantKeyStore(String keyStoreName) throws Exception {
+
+        if (isCachedKeyStoreValid(keyStoreName)) {
+            return tenantKeyStores.get(keyStoreName).getKeyStore();
+        }
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
```

Comment:
Shall we verify the `keyStoreName` not empty? In some cases if the tenantKeystore is gets deleted, after logged into the tenant there should be a null value. IN that case, path variable will not be defined as expected. Further its better to add some debug/info level logs as well.

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/938baad11ff3d0027a8299b50a1255950e925604.
(Added null checks to the public getKeyStore method)